doc/development/guide/documentation.rst

```diff
@@ -1,39 +1,449 @@
+  section is included.
+
+* **Do not list methods and properties.** Class properties and methods will be automatically
+  listed alongside the class in the generated documentation.
+
+* **List attributes if relevant.**
+
+Document all methods and properties as you would for functions above. However, there are two
+exemptions, where docstrings should *not* be provided:
+
+* **Magic or special methods.** This includes methods such as ``__init__``, ``__call__``,
+  ``__len__``, ``__getitem__``, etc. Only provide docstrings for special methods if
+  their behaviour is modified in a non-standard way (e.g., if negative indices are allowed
+  as arguments for ``__getitem__``).
+
+* **Overwritten methods.** Overwritten inherited methods automatically inherit the parent methods
+  docstring, even when overwritten. This is particularly useful when creating an instance
+  of an abstract base class; the abstract method docstring is defined in the parent, and automatically
+  inherited by the child class, even when overwritten.
+
+  As with special methods, only provide docstrings for overwritten methods if
+  their behaviour is modified in a non-standard way (as defined in the parent class).
+
+
+Variables
+~~~~~~~~~
+
+Module level variables may also be optionally documented, by providing a triple-quote docstring
+**below** the variable definition. For example,
+
+.. code-block:: python
+
+    x = {"John": 23, "James": 54}
+    """dict[str, int]: stores the ages of known users"""
+
+The syntax is the same as those used for describing function arguments.
+
+
+.. _code_examples:
+
+Code examples
+-------------
+
+Code examples are very important; they *show* readers how the function or class should be used.
+When writing code examples for docstrings, use the following guidelines:
+
+- You may assume that PennyLane is imported as ``qml`` and NumPy is imported as ``np`` in the code examples.
+  All other imports must be specified explicitly.
+
+- For single line statements and associated output, use Python console syntax (``pycon``):
+
+  .. code-block:: rest
+
+      >>> circuit(0.5, 0.1)
+      [0.43241234312, -0.543534534]
+
+  Multi-line statements should use ``...`` to indicate continuation lines:
+
+  .. code-block:: rest
+
+      >>> dev = qml.device("default.qubit", wires=1)
+      >>> @qml.qnode(dev)
+      >>> def circuit(x):
+      ...     qml.RX(x, wires=0)
+      ...     return qml.expval(qml.PauliZ(0))
+      >>> circuit(0.5)
+      0.8775825618903726
+
+  For larger, more complicated code blocks, favour standard Python code-block with
+  Python console syntax for displaying output:
+
+  .. code-block:: rest
+
+      .. code-block:: python3
+
+          dev = qml.device("default.qubit", wires=1)
+          @qml.qnode(dev)
+          def circuit(x):
+              qml.RX(x, wires=0)
+              return qml.expval(qml.PauliZ(0))
+
+      Executing this circuit:
```

Comment:
I'm a bit hesitent to recommend this as a standard. It is redundant, and it led to a lot of pollution in the changelog of last version. 
How do you feel about concatenating these two code blocks? (so `>>>` is used in parts but not all)?

Reply:
I don't think this is possible, these are two separate pygments lexers (`python` and `pycon` respectively). In the best case scenario it will build but the highlighting on the output lines will be incorrect. Worst case scenario the docs will error on building.

There have been a few cases I've had to 'split up' python and pycon syntax into separate blocks because it was breaking the docs.

Reply:
Should we have an exception for the changelog? No Python code blocks, just pycon?

Reply:
This one I'm uncertain about. It feels like we're often adding unnecessary text (e.g., "Executing this circuit") not because it's useful, but just to work around quirks of the pygments lexers. 

It's not just the changelog, it appears in the docstrings as well sometimes. I'd prefer if we had a straightforward working procedure that didn't rely on the current pattern of adding unecessary half-sentences just to break up the code blocks to use two different lexers.

Reply:
Upon retrospection, likely this means we should use the pycon syntax preferentially

Reply:
> Upon retrospection, likely this means we should use the pycon syntax preferentially

I concur.

There are a few cases where longer code examples are given in the `.. UsageDetails::` directive; examples that come to mind are the templates and QNN layers. Using `pycon` for these larger examples tends to reduce readability, since `pycon` doesn't allow linebreaks/comments are harder to include. I think these are the only places where `python` > `pycon`.

Reply:
c.f. https://pennylane.readthedocs.io/en/stable/code/api/pennylane.broadcast.html#usageDetails